src/connection.c

```diff
@@ -244,3 +244,24 @@ int connControl(ConnectionType *ct, unsigned long cmd, unsigned long arg0, unsig
         return ct->control(cmd, arg0, arg1, arg2, arg3);
     return C_ERR;
 }
+
+sds getListensInfoString(sds info) {
+    for (int j = 0; j < CONN_TYPE_MAX; j++) {
+        connListener *listener = &server.listeners[j];
+        if (listener->ct == NULL)
+            continue;
+
+        info = sdscatfmt(info, "%s:type=%s", listener->ct->get_type(NULL),
+               listener->ct->extension_mode ? "extension" : "builtin");
+        for (int i = 0; i < listener->count; i++) {
+            info = sdscatfmt(info, ",bind=%s", listener->bindaddr[i]);
+        }
+
+        if (listener->port)
+            info = sdscatfmt(info, ",port=%i", listener->port);
```

Comment:
I was thinking of something like this (to show the list of loaded extensions):
```
extensions:tls,rdma
```

i can see how what you did can be useful too, so i'm not sure which one we prefer, or maybe both?
but if we do keep it, i think the info field name should be prefixed (i don't think we want the info name to be the plain extension name), e.g:
```
listener_tcp:type=builtin,bind=127.0.0.1,port=6380
listener_unix:type=builtin,bind=/run/redis.sock
listener_tls:type=extension,bind=127.0.0.1,port=6379
```
@yossigo WDYT?

Reply:
I also noticed `tcp_port:6379` in the `INFO SERVER` command. And the code `server.port ? server.port : server.tls_port` means `TCP` has higher priority than `TLS`. If we can ignore the backwards capability in this output, this field can be removed too.

Reply:
no, i think we need to maintain backwards compatibility.
p.s. i did discus this topic with Yossi today.
he suggested that the info keyword will be index based, could be easier for clients to enumerate on them (like we list slaves and databases).
i.e.
```
listener0:name=tcp,type=builtin,bind=127.0.0.1,port=6380
listener1:name=unix,type=builtin,bind=/run/redis.sock
listener2:name=tls,type=extension,bind=127.0.0.1,port=6379
```
i'm also not sure we need the `type=` part, since i do think we need another place that lists extensions, so i suggest to drop it.
i.e. in the future we'll have extensions for compression algorithms, or persistence volume access, not just listeners, so extensions should be listed separately, not as part of the listeners list.

Yossi suggested that the extensions be listed in a section of their own like modules, one extension per line, rather than a single info comma separated field i suggested.

But it looks like we didn't come to an agreement yet about whether extensions are a type of module or not, Yossi argues that they should be just another type of modules), and in that case we don't need to list them them at all, they'll be listed in the modules section.

so i'll ask that you'll wait a few more days until we decide that, but meanwhile you can fix the `listener` info lines as i described above.